src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs

```diff
@@ -289,13 +300,29 @@ public bool TryGetHasUnscopedRefAttribute(out bool hasUnscopedRefAttribute)
                 isRuntimeSpecialName: (mdFlags & PropertyAttributes.RTSpecialName) != 0,
                 callMethodsDirectly);
 
-            static bool anyUnexpectedRequiredModifiers(ParamInfo<TypeSymbol>[] propertyParams)
+            static bool anyUnexpectedRequiredModifiers(ParamInfo<TypeSymbol>[] propertyParams, bool isExtensionInstanceUnderlyingSymbol)
             {
-                return propertyParams.Any(p => (!p.RefCustomModifiers.IsDefaultOrEmpty && p.RefCustomModifiers.Any(static m => !m.IsOptional && !m.Modifier.IsWellKnownTypeInAttribute())) ||
-                                               p.CustomModifiers.AnyRequired());
+                for (int i = 0; i < propertyParams.Length; i++)
+                {
+                    var p = propertyParams[i];
+
+                    if (!p.RefCustomModifiers.IsDefaultOrEmpty && p.RefCustomModifiers.Any(static m => !m.IsOptional && !m.Modifier.IsWellKnownTypeInAttribute()))
+                    {
+                        return true;
+                    }
+
+                    if ((!isExtensionInstanceUnderlyingSymbol || i != 1) && p.CustomModifiers.AnyRequired())
```

Comment:
Should it be `i != 0`?

Reply:
> Should it be `i != 0`?

No, an information for return type comes at position 0.